sdk/identity/src/token_credentials/managed_identity_credentials.rs

```diff
@@ -22,7 +22,7 @@ pub struct ManagedIdentityCredential;
 impl TokenCredential for ManagedIdentityCredential {
     async fn get_token(&self, resource: &str) -> Result<TokenResponse, AzureError> {
         let msi_endpoint = std::env::var(MSI_ENDPOINT_ENV_KEY)
-            .unwrap_or("http://169.254.169.254/metadata/identity/oauth2/token".to_owned());
+            .unwrap_or_else(|_| "http://169.254.169.254/metadata/identity/oauth2/token".to_owned());
```

Comment:
Maybe I'm missin something, but is there a need for a closure here? We know the value ahead of time, no?

Reply:
This saves an allocation even the variable is set. How it was originally written we would allocate a string even if we never used it. Small optimization, but why not?

Reply:
Ah right. Sure, LGTM.

Reply:
Yeah, it is a `clippy` lint warning. I started going through them and then realized it wouldn't be the right PR to fix them all, so I stopped after a few.